numpy/core/src/umath/scalarmath.c.src

```diff
@@ -151,16 +157,16 @@ static void
  *         npy_long, npy_ulong, npy_longlong, npy_ulonglong#
  * #neg = (1,0)*5#
  */
-static void
+static NPY_INLINE int
 @name@_ctype_divide(@type@ a, @type@ b, @type@ *out) {
     if (b == 0) {
-        npy_set_floatstatus_divbyzero();
         *out = 0;
+        return NPY_FPE_DIVIDEBYZERO;
     }
 #if @neg@
     else if (b == -1 && a < 0 && a == -a) {
-        npy_set_floatstatus_overflow();
         *out = a / b;
+        return NPY_FPE_OVERFLOW;
```

Comment:
Help me out: when is `a < 0 && a == -a` for bytes?

Reply:
Oh, `abs(np.int(-128)) == -128`.